libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -1138,6 +1130,28 @@ private async Task<AppCredentials> GetAppCredentialsAsync(string appId, string o
             return appCredentials;
         }
 
+        /// <summary>
+        /// Gets the AppId of the Bot out of the TurnState.
+        /// </summary>
+        /// <param name="turnContext">The context object for the turn.</param>
+        /// <returns>Bot's AppId.</returns>
+        private string GetBotAppId(ITurnContext turnContext)
+        {
+            var botIdentity = (ClaimsIdentity)turnContext.TurnState.Get<IIdentity>(BotIdentityKey);
+            if (botIdentity == null)
+            {
+                throw new InvalidOperationException("An IIdentity is required in TurnState for this operation.");
+            }
+
+            var appId = botIdentity.Claims.FirstOrDefault(claim => claim.Type == AuthenticationConstants.AudienceClaim)?.Value;
```

Comment:
Not sure this is correct, we have a helper in JwtTokenValidation called GetAppIdFromClaims that uses AppId ot Azp depending of the version of the token (v1 or v2), I am not clear what value will this have for websockets but based on some tests I've done for non websocket scenarios the audience claim doesn't always have the AppId.
@carloscastro to confirm